test/extensions/filters/http/composite/composite_filter_integration_test.cc

```diff
@@ -1,49 +1,130 @@
+#include "envoy/matcher/matcher.h"
+#include "envoy/type/matcher/v3/http_inputs.pb.h"
+#include "envoy/type/matcher/v3/http_inputs.pb.validate.h"
+
+#include "source/common/http/match_delegate/config.h"
 
+#include "test/common/http/common.h"
+#include "test/integration/filters/add_body_filter.pb.h"
+#include "test/integration/filters/set_response_code_filter_config.pb.h"
 #include "test/integration/http_integration.h"
 #include "test/mocks/http/mocks.h"
 #include "test/test_common/utility.h"
 
+#include "absl/strings/str_format.h"
 #include "gtest/gtest.h"
 
 namespace Envoy {
 
+namespace {
+
+using envoy::config::route::v3::Route;
+using envoy::config::route::v3::VirtualHost;
+using envoy::extensions::common::matching::v3::ExtensionWithMatcherPerRoute;
+using envoy::extensions::filters::common::matcher::action::v3::SkipFilter;
+using envoy::extensions::filters::http::composite::v3::ExecuteFilterAction;
+using Envoy::Protobuf::MapPair;
+using Envoy::ProtobufWkt::Any;
+using envoy::type::matcher::v3::HttpRequestHeaderMatchInput;
+using test::integration::filters::AddBodyFilterConfig;
+using test::integration::filters::SetResponseCodeFilterConfig;
+using xds::type::matcher::v3::Matcher_OnMatch;
+
 class CompositeFilterIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
                                        public HttpIntegrationTest {
 public:
   CompositeFilterIntegrationTest() : HttpIntegrationTest(Http::CodecType::HTTP1, GetParam()) {}
 
-  void initialize() override {
-    config_helper_.prependFilter(R"EOF(
-  name: composite
-  typed_config:
-    "@type": type.googleapis.com/envoy.extensions.common.matching.v3.ExtensionWithMatcher
-    extension_config:
-      name: composite
+  ExtensionWithMatcherPerRoute createPerRouteConfig(
+      std::function<void(envoy::config::core::v3::TypedExtensionConfig*)> base_action_function) {
+
+    ExtensionWithMatcherPerRoute per_route_config;
+    auto matcher_tree = per_route_config.mutable_xds_matcher()->mutable_matcher_tree();
+
+    auto matcher_input = matcher_tree->mutable_input();
+    matcher_input->set_name("request-headers");
+    HttpRequestHeaderMatchInput match_input;
+    match_input.set_header_name("match-header");
+    matcher_input->mutable_typed_config()->PackFrom(match_input);
+
+    auto map = matcher_tree->mutable_exact_match_map()->mutable_map();
+    Matcher_OnMatch match;
+    auto mutable_action = match.mutable_action();
+    mutable_action->set_name("composite-action");
+
+    ExecuteFilterAction filter_action;
+    base_action_function(filter_action.mutable_typed_config());
+    mutable_action->mutable_typed_config()->PackFrom(filter_action);
+
+    (*map)["match"] = match;
+    return per_route_config;
+  }
+
+  void addPerRouteResponseCodeFilter(const std::string& filter_name,
+                                     const std::string& route_prefix, const int& code,
+                                     const bool& response_prefix = false) {
```

Comment:
tiny nit : `const bool& response_prefix` ---> `bool response prefix` (go/c-readability-advice#value-param)
No need to change in this PR.

Reply:
Done.